nav2_collision_monitor/src/collision_detector_node.cpp

```diff
@@ -307,10 +307,22 @@ void CollisionDetector::process()
   // Points array collected from different data sources in a robot base frame
   std::vector<Point> collision_points;
 
+  std::unique_ptr<nav2_msgs::msg::CollisionDetectorState> state_msg =
+    std::make_unique<nav2_msgs::msg::CollisionDetectorState>();
+
   // Fill collision_points array from different data sources
   for (std::shared_ptr<Source> source : sources_) {
     if (source->getEnabled()) {
-      source->getData(curr_time, collision_points);
+      if (!source->getData(curr_time, collision_points) &&
+        source->getSourceTimeout().seconds() != 0.0)
```

Comment:
It is a `double` behind, it will be fine (apart if we want a very accurate and not small `source_timeout` like 10000.0000000001). `nanoseconds() `returns a `rcl_duration_value_t`, more messy

Reply:
@doisyg just checking -- if you've tested this and you know it works, I'll trust you. The `count()` though makes me highly suspicious though, since chrono's count usually returns a whole numbered count from my experience:

```
  return std::chrono::duration<double>(std::chrono::nanoseconds(rcl_duration_.nanoseconds)).count();
```

But the `duration` might do differently. I both love/hate chrono. Very useful, but also the API isn't very consistent at times

Reply:
> @doisyg just checking 

As you should, I get confused very often with all these slightly different Time API.

> if you've tested this and you know it works

Confirming that it is tested and working